### ISSUE TITLE
cannot create a generic field with specified type

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#9114]
### Summary of Problem
The compiler issues an error when trying to compile code using a field which is generic but has a declared type.

### Steps to Reproduce

**Source Code:**

```chapel
class MyClass {
  var x;
}

record R {
  var x:MyClass;
}

proc test() {
  var r = new R(new MyClass(1));
  writeln(r);
}
test();
```

Results in
```
def.chpl:7: error: unresolved type specifier '_type_construct_MyClass()'
def.chpl:2: note: candidates are: _type_construct_MyClass(type x) [266970]
def.chpl:7: note: unresolved call had id 1153228
```

Here is a similar example with `numeric`:

``` chapel
record R {
  var x:numeric;
}

proc test() {
  var r = new R(1);
  writeln(r);
}
test();
```

results in `error: invalid generic type specification on class field`.

**Associated Future Test(s):**

None yet, as far as I know.

**Comment by bradcray** [comment]
Hi @mppf:  I'm wondering if this is resolved now.  The following program worked for me:

```chapel
pragma "use default init"
class MyClass {
  var x;
}

pragma "use default init"
record R {
  var x:unmanaged MyClass;
}

proc test() {
  var r = new R(new unmanaged MyClass(1));
  writeln(r);
}
test();
```

If not, would you file a future against this issue?

**Comment by mppf** [comment]
This was fixed in PR  #9489 but the fix requires that initializers are used (it doesn't work for constructors).
